### PR TITLE
[TT-8498] Add hybrid pump as a default for mdcb dp charts

### PR DESCRIPTION
**Pull request by zalbiraw**
## Description

- Add support of hybrid pump
- Enable hybrid pump in mdcb-data-plane by default.

## Related Issue

[TT-8498](https://tyktech.atlassian.net/browse/TT-8498)

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Refactoring or add test (improvements in base code or adds test coverage to functionality)
- [ ] Documentation updates or improvements.

## Checklist

- [x] Make sure you are requesting to **pull a topic/feature/bugfix branch** (right side). If PRing from your fork, don't come from your `master`!
- [x] Make sure you are making a pull request against our **`master` branch** (left side). Also, it would be best if you started *your change* off *our latest `master`*.
- [ ] My change requires a change to the documentation.
     - [ ] I have manually updated the README(s)/documentation accordingly.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

[TT-8498]: https://tyktech.atlassian.net/browse/TT-8498?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by joshblakeley** [comment]
@caroltyk this is really badly need can we merge please

**Comment by caroltyk** [comment]
@joshblakeley There's some more work required on this, @komalsukhani is already on it.

File: components/tyk-pump/templates/deployment-pmp.yaml [hunk 1]
```diff
@@ -53,8 +53,46 @@ spec:
           # Legacy support for Redis Cluster driver. Driver dropped in v3.0.0.
           - name: REDIGOCLUSTER_SHARDCOUNT
             value: "128"
-
-         {{ if has "prometheus" .Values.pump.backend }}
+         {{ if has "hybrid" .Values.pump.backend }}
+          # Configure Hybrid Pump for Tyk Pump
+          - name: TYK_PMP_PUMPS_PROMETHEUS_TYPE
```
Comment (komalsukhani) [inline components/tyk-pump/templates/deployment-pmp.yaml:58]:
```suggestion
          - name: TYK_PMP_PUMPS_HYBRID_TYPE
```

File: components/tyk-pump/templates/deployment-pmp.yaml [hunk 2]
```diff
@@ -53,8 +53,46 @@ spec:
...
+{{ .Values.pump.hybridPump.userApiKey | toYaml | indent 12 }}
+            {{ end }}
+          - name: TYK_PMP_PUMPS_HYBRID_META_AGGREGATED
+            value: "{{ .Values.pump.hybridPump.enableAggregateAnalytics }}"
+          - name: TYK_PMP_PUMPS_HYBRID_META_USESSL
+            value: "{{ .Values.pump.hybridPump.useSSL }}"
+          - name: TYK_PMP_PUMPS_HYBRID_META_SSLINSECURESKIPVERIFY
+            value: "{{ .Values.pump.hybridPump.skipSSLVerification }}"
+          - name: TYK_PMP_PUMPS_HYBRID_META_GROUPID
+            {{ if typeIs "string" .Values.pump.hybridPump.userApiKey }}
```
Comment (komalsukhani) [inline components/tyk-pump/templates/deployment-pmp.yaml:85]:
```suggestion
            {{ if typeIs "string" .Values.pump.hybridPump.groupID }}
```

File: components/tyk-pump/templates/deployment-pmp.yaml [hunk 3]
```diff
@@ -53,8 +53,46 @@ spec:
...
+            value: "{{ .Values.pump.hybridPump.userApiKey }}"
```
Comment (komalsukhani) [inline components/tyk-pump/templates/deployment-pmp.yaml:86]:
```suggestion
            value: "{{ .Values.pump.hybridPump.groupID }}"
```

File: components/tyk-pump/templates/deployment-pmp.yaml [hunk 4]
```diff
@@ -53,8 +53,46 @@ spec:
...
+            {{ else }}
+{{ .Values.pump.hybridPump.userApiKey | toYaml | indent 12 }}
```
Comment (komalsukhani) [inline components/tyk-pump/templates/deployment-pmp.yaml:88]:
```suggestion
{{ .Values.pump.hybridPump.groupID | toYaml | indent 12 }}
```

File: tyk-mdcb-data-plane/values.yaml [hunk 5]
```diff
@@ -384,11 +384,53 @@ tyk-pump:
...
+            name: secrets-tyk-data-plane
+            key: orgID
+      # API key of your dashboard user
+      userApiKey:
+        valueFrom:
+          secretKeyRef:
+            name: secrets-tyk-data-plane
+            key: apiKey
+      # needed in case you want to have multiple data-planes connected to the same redis instance
+      # TODO: this is a pump bug
```
Comment (komalsukhani) [inline tyk-mdcb-data-plane/values.yaml:416]: I tested with pump v1.8.0. It works when I set groupID

**Review by andrei-tyk — APPROVED** [review]

**Review by andrei-tyk — APPROVED** [review]

File: tyk-mdcb-data-plane/values.yaml [hunk 6]
```diff
@@ -368,23 +284,34 @@ tyk-pump:
...
       # tag for Tyk pump
-      tag: v1.7.0
+      tag: v1.8.0
 
       # image pull policy
       pullPolicy: IfNotPresent
 
     # backend defines the pumps to be created by default, as an array of string.
-    # Supported backends are ["mongo", "postgres", "prometheus"]
+    # Supported backends are ["mongo", "postgres", "prometheus","hbyrid"]
```
Comment (caroltyk) [inline tyk-mdcb-data-plane/values.yaml:293]:
```suggestion
    # Supported backends are ["mongo", "postgres", "prometheus","hybrid"]
```

File: tyk-mdcb-data-plane/values.yaml [hunk 7]
```diff
@@ -368,23 +284,34 @@ tyk-pump:
...
     # If you would like to use other backends such as ElasticSearch, please
     # configure the backend via environment variables.
     # By default, Tyk Pump uses Prometheus.
```
[the remark this answers is not in the document]
Comment (komalsukhani) [reply tyk-mdcb-data-plane/values.yaml]: Removed in [2eb9830](https://github.com/TykTechnologies/tyk-charts/pull/34/commits/2eb9830a71c65a72d78e04470eb65757c62aecc7)

**Pull request merged**